lib/solidus_feeds/generators/google_merchant.rb

```diff
@@ -0,0 +1,93 @@
+# lib/google_merchant.rb
+
+require 'builder'
+
+module SolidusFeeds
+  module Generators
+    # The GoogleMerchant XML feed as described in https://support.google.com/merchants/answer/7052112.
+    class GoogleMerchant
+      attr_accessor :products, :host
+
+      def initialize(products, host:)
+        self.products = products
+        self.host = host
+      end
+
+      def call(io)
+        builder = Builder::XmlMarkup.new(target: io, indent: 0)
+        render_template(builder)
+      end
+
+      private
+
+      def render_template(xml)
+        xml.rss version: "2.0", "xmlns:g" => "http://base.google.com/ns/1.0" do
+          xml.channel do
+            xml.title 'store title'
+            xml.link 'store link'
+            xml.description 'store description'
+            xml.language 'store lang'
+            products.find_each do |product|
+              xml.item do
+                xml.tag! 'g:id', product.id
+                xml.tag! 'g:title', product.name
+                xml.tag! 'g:description', product.description
+                xml.tag! 'g:link', link(product)
+                xml.tag! 'g:image_link', image_link(product)
+                xml.tag! 'g:condition', condition
+                xml.tag! 'g:price', price(product)
+                xml.tag! 'g:availability', availability(product)
+                xml.tag! 'g:brand', brand
+                xml.tag! 'g:mpn', mpn(product)
+                xml.tag! 'g:google_product_category', category
+              end
+            end
+          end
+        end
+      end
+
+      def link(product)
+        spree_routes.product_url(product, host: host)
+      end
+
+      def spree_routes
+        @spree_routes ||= Spree::Core::Engine.routes.url_helpers
+      end
+
+      def image_link(product)
+        return unless product.images.any?
+
+        attachment_url = product.images.first.attachment.url(:large)
+        asset_host = ActionController::Base.asset_host
+
+        URI.join(asset_host, attachment_url)
+      end
+
+      # Must be "new", "refurbished", or "used".
+      def condition
```

Comment:
This should receive the product as an argument, I think. That way, it can be overridden to return the condition depending on the product.